### ISSUE TITLE
[BUG][Python] Generated Python Code with `oneOf` and `required` is incorrect

### ISSUE THREAD

**Issue by andrew-matteson** [OpenAPITools/openapi-generator#1920]
#### Bug Report Checklist

- [x] Have you provided a full/minimal spec to reproduce the issue?
- [x] Have you validated the input using an OpenAPI validator ([example](https://apidevtools.org/swagger-parser/online/))?
- [x] What's the version of OpenAPI Generator used?
- [x] Have you search for related issues/PRs?
- [x] What's the actual output vs expected output?
- [ ] [Optional] Bounty to sponsor the fix ([example](https://www.bountysource.com/issues/66123212-javascript-client-produces-a-wrong-object-for-a-string-enum-type-that-is-used-with-ref))

##### Description

Generated Python code includes `ValueError` where it shouldn't

Generated code:
```
    @list.setter
    def list(self, list):
        """Sets the list of this Distribution.


        :param list: The list of this Distribution.  # noqa: E501
        :type: str
        """
        if list is None:
            raise ValueError("Invalid value for `list`, must not be `None`")  # noqa: E501

        self._list = list
```

But this setter is invalid when applied to `Distribution` or `GridDistribution` from the example below.

##### openapi-generator version
Built from most recent master 963c0028302e2edb02a93fd5bee4fb8d0a9c002b

##### OpenAPI declaration file content or url
```
openapi: "3.0.0"
info:
  version: 1.0.0
  title: Batch Simulator
servers:
  - url: /foo/
paths:
  /foo/:
    get:
      summary: Dummy
      description: dummy
      responses:
        '200':
          description: dummy
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/distribution'

components:
  schemas:
    distribution:
      description: An abstract distribution
      type: object
      additionalProperties: false
      properties:
        dummy:
          type: string # Without this "Distribution" isn't generatred because "it's a free-form object"
      discriminator:
        propertyName: class_name
      oneOf:
        - $ref: '#/components/schemas/list-distribution'
        - $ref: '#/components/schemas/grid-distribution'

    list-distribution:
      description: version 1
      type: object
      properties:
        class_name:
          type: string
        list:
          type: string
      required:
        - list # This gets applied to distribution, when it should only be applied to list-distribution

    grid-distribution:
      description: version 2
      properties:
        class_name:
          type: string
```

##### Command line used for generation

`java -jar ~/Repos/openapi-generator/modules/openapi-generator-cli/target/openapi-generator-cli.jar  generate -g python -i test.yaml -o test_client`

##### Steps to reproduce

1. Build from most recent master (or grab appropriate snapshot)
2. Run CLI generator on the provided minimal example
3. Inspect `openapi_client.models.distribution` and look at the generated setter for `list`. This should not raise a `ValueError` because this is applied to all objects of the `Distribution` before the final class is resolved.

##### Related issues/PRs
#1662 

##### Suggest a fix
The `ValueError` should only be applied when generating the final classes, in this case `ListDistribution`. I don't have any ideas about how to address this.

**Comment by fsonntag** [comment]
@andrew-matteson Have you found a way to circumvent this?

**Comment by spacether** [comment]
This is happening because the java code is loading required oneOf properties into the composed schema distribution @wing328 
@andrew-matteson your schema for distribution is incorrect. It must have additionalProperties set to true. additionalProperties only describes properties that are not in the current schema's properties. If the current schema has not properties then all properties are additional. oneOf/anyOf/allOf schemas are not included in properties/additionalProperties logic.

**Comment by wing328** [comment]
```
> components:
  schemas:
    distribution:
      description: An abstract distribution
      type: object
      additionalProperties: false
      properties:
        dummy:
          type: string # Without this "Distribution" isn't generatred because "it's a free-form object"
      discriminator:
        propertyName: class_name
      oneOf:
        - $ref: '#/components/schemas/list-distribution'
        - $ref: '#/components/schemas/grid-distribution'
```

Please remove the`dummy` property as the latest python client has better support for oneOf and it will no longer treat it as a free-form object.

**Comment by spacether** [comment]
This was fixed in openapi-generator v6.2.0 which allows oneOf + discriminator + properties + required all in one schema
